CAN_Logger_with_Autobaud_and_Requests_no_CAN2_AES_CBC_SHA256/CAN_Logger_with_Autobaud_and_Requests_no_CAN2_AES_CBC_SHA256.ino

```diff
@@ -989,46 +1006,140 @@ void setup(void) {
-  baudFile.open("baudRate.txt", O_RDWR | O_CREAT | O_AT_END);
-  sprintf(timeString,"%04d-%02d-%02dT%02d:%02d:%02d,%d,%d",year(),month(),day(),hour(),minute(),second(),Can0.baud_rate,Can1.baud_rate);
-  baudFile.print(timeString);
-  baudFile.print(" TU");
-  baudFile.print(logger_name);
-  baudFile.print(current_file);
-  baudFile.print(" IV: ");
-  for (int n = 0; n < sizeof(init_vector); n++){
-    baudFile.print(",0x");
-    if (init_vector[n] < 16){
-      baudFile.print("0");
-    }
-    
-    baudFile.print(init_vector[n],HEX);
-  }
-  baudFile.print(" KEY:");
-    for (int n =0; n < sizeof(aeskey);n++){
-    baudFile.print(",0x");
-    if (aeskey[n] <16){
-      baudFile.print("0");
-    }
-     baudFile.print(aeskey[n],HEX);
-  }
-  baudFile.println("");
-  baudFile.close();
-  Serial.println("Done.");
-
   
   sprintf(file_name_prefix,"%s%s",brand_name,logger_name);
   Serial.print("The filename prefix is ");
   Serial.println(file_name_prefix);
+
+  
   
   recording = true;
   pinMode(POWER_PIN,INPUT_PULLUP);
   attachInterrupt(digitalPinToInterrupt(POWER_PIN), close_binFile, RISING);
 }
 
+
+void write_final_meta_data(){
+  sha256Instance->update(cipher_text,BUFFER_SIZE);
+  sha256Instance->final(hash);
+  delete sha256Instance;
+  Serial.print(",BIN-SHA:");
+  memcpy(&data_file_contents[data_file_index],",BIN-SHA:",9);
+  data_file_index+=9;
+  for (int n = 0; n < sizeof(hash); n++){
+    char hex_digit[3];
+    sprintf(hex_digit,"%02X",hash[n]);
+    Serial.print(hex_digit);
+    memcpy(&data_file_contents[data_file_index],&hex_digit,2);
+    data_file_index+=2;
+  }
+    
+  sha256Instance=new Sha256();
+  sha256Instance->update(data_file_contents, strlen((const char*)data_file_contents));
+  sha256Instance->final(hash);
+  delete sha256Instance;
+  
+  Serial.print(",TXT-SHA:");
+  memcpy(&data_file_contents[data_file_index],",TXT-SHA:",9);
+  data_file_index+=9;
+  for (int n = 0; n < sizeof(hash); n++){
+    char hex_digit[3];
+    sprintf(hex_digit,"%02X",hash[n]);
+    Serial.print(hex_digit);
+    memcpy(&data_file_contents[data_file_index],&hex_digit,2);
+    data_file_index+=2;
+  }
+  
+  // Compute Signature here
+  atecc.createSignature(hash);
+  Serial.print(",SIG:");
+  memcpy(&data_file_contents[data_file_index],",SIG:",5);
+  data_file_index+=5;
+  
+  for (int n = 0; n < sizeof(atecc.signature); n++){
+    char hex_digit[3];
+    sprintf(hex_digit,"%02X",atecc.signature[n]);
+    Serial.print(hex_digit);
+    memcpy(&data_file_contents[data_file_index],&hex_digit,2);
+    data_file_index+=2;
+  }
+  Serial.println();
+
+  //Write the metadata file
+  sprintf(data_file_name,"%s%s%s.txt",brand_name,logger_name,current_file);
+  if (!dataFile.open(data_file_name, O_RDWR | O_CREAT)) {
+    YELLOW_LED_fast_blink_state == true;
+    Serial.println("Error opening ");
+    Serial.println(data_file_name);
+  }
+  else{
+    for (int i = 0; i<strlen((const char*)data_file_contents);i++){
```

Comment:
"for (int i = 0; i<strlen((const char*)data_file_contents);i++){"
will become 
"for (int i = 0; i < strlen((const char*)data_file_contents); i++){"